packages/core/src/routes/resource.test.ts

```diff
@@ -0,0 +1,117 @@
+import { Resource, CreateResource } from '@logto/schemas';
+
+import { mockResource, mockScope } from '@/utils/mock';
+import { createRequester } from '@/utils/test-utils';
+
+import resourceRoutes from './resource';
+
+jest.mock('@/queries/resource', () => ({
+  findTotalNumberOfResources: jest.fn(async () => ({ count: 10 })),
+  findAllResources: jest.fn(async (): Promise<Resource[]> => [mockResource]),
+  findResourceById: jest.fn(async (): Promise<Resource> => mockResource),
+  insertResource: jest.fn(
+    async (body: CreateResource): Promise<Resource> => ({
+      ...mockResource,
+      ...body,
+    })
+  ),
+  updateResourceById: jest.fn(
+    async (_, data: Partial<CreateResource>): Promise<Resource> => ({
+      ...mockResource,
+      ...data,
+    })
+  ),
+  deleteResourceById: jest.fn(),
+}));
+
+jest.mock('@/queries/scope', () => ({
+  findAllScopesWithResourceId: jest.fn(async () => [mockScope]),
+}));
+
+jest.mock('@/utils/id', () => ({
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  buildIdGenerator: jest.fn(() => () => 'randomId'),
+}));
+
+describe('resource routes', () => {
+  const resourceRequest = createRequester(resourceRoutes);
+
+  it('GET /resources', async () => {
+    const response = await resourceRequest.get('/resources');
+    expect(response.status).toEqual(200);
```

Comment:
it seems that `supertest` will throw an error on non 2xx response. And can we use request chains like that: https://github.com/visionmedia/supertest#example

Reply:
not sure, but using the async/await might be consistent with our code style?

<img width="682" alt="image" src="https://user-images.githubusercontent.com/36393111/154885718-52034517-30e6-4231-91d9-1118d9e84ef0.png">

Did have some exception request cases listed below, and seems the request itself does not throw out the error.  I have to use resolve to get the response. 

<img width="707" alt="image" src="https://user-images.githubusercontent.com/36393111/154885862-cdda46d5-60cc-4e43-bbe8-50ec0cc21ed7.png">